src/renderer/task-runner.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  BisectRequest,
+  ElectronReleaseChannel,
+  RunResult,
+  RunnableVersion,
+  SetFiddleOptions,
+  SetupRequest,
+  TestRequest,
+} from '../interfaces';
+import { IpcEvents } from '../ipc-events';
+
+import { App } from './app';
+import { AppState } from './state';
+import { IpcRendererManager } from './ipc';
+import { Runner } from './runner';
+
+import { getVersionRange } from '../utils/get-version-range';
+import { normalizeVersion } from '../utils/normalize-version';
+
+export class TaskRunner {
+  private readonly autobisect: (v: RunnableVersion[]) => Promise<RunResult>;
+  private readonly done: (r: RunResult) => void;
+  private readonly hide: (channels: ElectronReleaseChannel[]) => Promise<void>;
+  private readonly log: (message: string) => void;
+  private readonly open: (o: SetFiddleOptions) => Promise<void>;
+  private readonly run: () => Promise<RunResult>;
+  private readonly setVersion: (ver: string) => Promise<void>;
+  private readonly show: (channels: ElectronReleaseChannel[]) => Promise<void>;
+
+  constructor(
+    app: App,
+    private readonly appState: AppState,
+    runner: Runner,
+    ipc: IpcRendererManager,
```

Comment:
How come we pass `ipcRendererManager` into the constructor here rather than importing it directly?

Reply:
Same reason that constructor DI is used for app et al, to reduce unnecessary dependencies on globals.  You could also ask why all these classes take appState as a constructor argument when it's also a global. *shrug* :slightly_smiling_face: 

Still, it's not a hill that I want to die on. window.ElectronFiddle.app isn't going anywhere. I don't have a commit ready to push for the DI removal but I'll get that to you ASAP; I want to push the other commits to resolve the other conversations.

Reply:
constructor DI removed in e89536e82ff48932d1b142fa35b897735c7c7190